rfc/0023-kubewarden-stack-telemetry.md

```diff
@@ -0,0 +1,373 @@
+- `value`: A field always set to `1`,
+  [added](https://github.com/longhorn/upgrade-responder/blob/a6f6c7736b7e420b07ae7d813765dac778ebc638/upgraderesponder/service.go#L560)
+  automatically by `upgrade-responder`.
+
+### Understanding Tags and Fields
+
+There are differences from InfluxDB `tags` and `fields` which may be relevant
+to document here as well. Tags are indexed metadata which can be used to group
+and filter data points. While `fields` are the actual metric data.
+
+| Aspect          | Tags                                                                                | Fields                                                                |
+| :-------------- | :---------------------------------------------------------------------------------- | :-------------------------------------------------------------------- |
+| **Purpose**     | Serve as indexed metadata used for filtering and grouping (e.g., `host`, `region`). | Hold the actual measured values (e.g., `temperature`, `cpu_usage`).   |
+| **Indexing**    | Stored in an inverted index; queries on tags are very fast.                         | Not indexed; queries must scan data, which can be slower.             |
+| **Cardinality** | High cardinality (many unique values) can degrade database performance.             | Low impact on performance; many distinct field values are acceptable. |
+| **Data Types**  | Limited to strings.                                                                 | Support numeric types, booleans, and strings.                         |
+
+### Grafana Dashboard
+
+Alongside the `upgrade-responder` and InfluxDB, a Grafana instance with a custom
+dashboard will be configured. The dashboard will use InfluxDB as its data
+source and provide queries to visualize the desired metrics.
+
+### Telemetry Server Infrastructure
+
+The `upgrade-responder`, InfluxDB, and Grafana instances used for Kubewarden
+telemetry will be maintained by the SUSE team. All members of the Kubewarden
+organization on GitHub will have access to the collected data.
+
+## Helm Chart Changes
+
+The Kubewarden Helm charts would require new values to configure the
+controller's CLI flags.
+
+```yaml
+stackTelemetry:
+  enabled: true
+  endpoint: "https://metrics.kubewarden.io"
+  period: "1h"
+```
+
+When `stackTelemetry.enabled` is `true`, the feature is enabled in the
+controller via the corresponding CLI flags.
+
+The `stackTelemetry.enabled` setting will be `true` by default. Users will be
+notified during the Kubewarden stack upgrade that this telemetry is included,
+and instructions to disable it will be provided
+
+# Drawbacks
+
+- When new metadata or more metric data is required, it necessitates code
+  changes and a new release for both the controller and, maybe, the telemetry
+  server. The JSON schema versions must be kept in sync.
+- We are solely responsible for ensuring secure communication (TLS,
+  authentication, etc.) with the remote telemetry server.
```

Comment:
Is that a drawback? I think this implicit with all telemetry solutions.

The TLS certificate used by the server is going to be issued by a trusted CA that is part of the standard `ca-certificates` bundles.

As for the authorization, this must be in place only for the grafana dashboard, but this will delegate authentication to GitHub